### ISSUE TITLE
🐛[Firebase_messaging]App configured using flutter CLI tool but console still shows `Could not locate configuration file: 'GoogleService-Info.plist'.` and `No app has been configured yet.` logs in the console.

### ISSUE THREAD

**Issue by cycloss** [FirebaseExtended/flutterfire#8119]
## Bug report

**Describe the bug**
I'm always getting the warning:

```
8.11.0 - [Firebase/Core][I-COR000012] Could not locate configuration file: 'GoogleService-Info.plist'.`
8.11.0 - [Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add `FirebaseApp.configure()` to your application initialization. This can be done in in the App Delegate's application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI). Read more: https://goo.gl/ctyzm8.
8.11.0 - [Firebase/Core][I-COR000005] No app has been configured yet.
```

### Steps to reproduce

Follow the documentation at https://firebase.flutter.dev/docs/overview/#initializing-flutterfire, and https://firebase.flutter.dev/docs/messaging/usage.

### Expected behavior

I'm using the automatic initialisation via dart with `flutterfire configure`, so as I understand I do not need the `GoogleService-Info.plist`. I don't understand why I'm still getting this message that the app has not been configured with I have followed the steps and my flutter `main` function includes initialisation:

```dart
void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
  runApp(const MyApp());
}
```

It's worth saying that iOS notifications are actually working, and I can send them to a real device from the fcm console. Are these warnings just a red herring?

**Comment by darshankawar** [comment]
@cycloss 
Can you provide your `flutter doctor -v` and plugin versions you are using ?
A similar issue was fixed but not sure if you are hitting the same case or not. https://github.com/FirebaseExtended/flutterfire/issues/7752

**Comment by cycloss** [comment]
@darshankawar sure:

```
[✓] Flutter (Channel stable, 2.10.1, on macOS 12.2 21D49 darwin-x64, locale en-GB)
    • Flutter version 2.10.1 at /usr/local/Caskroom/flutter/2.0.4/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db747aa133 (12 days ago), 2022-02-09 13:57:35 -0600
    • Engine revision ab46186b24
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/ted/Library/Android/sdk
    • Platform android-31, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.64.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • iPhone 11 (mobile) • AD20B769-F009-4772-AC51-36EB495E6530 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64     • macOS 12.2 21D49
      darwin-x64
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome
      98.0.4758.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

For my firebase plugins i'm using:

```
  firebase_core:
    dependency: "direct main"
    description:
      name: firebase_core
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.12.0"
  firebase_core_platform_interface:
    dependency: transitive
    description:
      name: firebase_core_platform_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "4.2.4"
  firebase_core_web:
    dependency: transitive
    description:
      name: firebase_core_web
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.5.4"
  firebase_messaging:
    dependency: "direct main"
    description:
      name: firebase_messaging
      url: "https://pub.dartlang.org"
    source: hosted
    version: "11.2.6"
  firebase_messaging_platform_interface:
    dependency: transitive
    description:
      name: firebase_messaging_platform_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "3.1.6"
  firebase_messaging_web:
    dependency: transitive
    description:
      name: firebase_messaging_web
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.2.7"
    ```

**Comment by it-hienvv** [comment]
I'm same

**Comment by darshankawar** [comment]
Thanks for the update. Using `firebase_messaging` plugin example and running with flutterfire configure settings, I do see the same logs in the console while building the app. Although the app compiles and builds properly, below logs are displayed:

<details>
<summary> console log messages </summary>

```

Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Warning: Missing build name (CFBundleShortVersionString).
Warning: Missing build number (CFBundleVersion).
Action Required: You must set a build name and number in the pubspec.yaml file version field before submitting to the App Store.
Running pod install...
Running Xcode build...
Xcode build done.                                           53.9s
8.9.0 - [Firebase/Core][I-COR000012] Could not locate configuration file: 'GoogleService-Info.plist'.
remote notifications are not supported in the simulator
Debug service listening on ws://127.0.0.1:51982/vW7B1vblqX8=/ws
Syncing files to device iPhone 12 Pro Max...
8.9.0 - [Firebase/Core][I-COR000005] No app has been configured yet.
8.9.0 - [Firebase/Core][I-COR000005] No app has been configured yet.
8.9.0 - [Firebase/Messaging][I-FCM002022] APNS device token not set before retrieving FCM Token for Sender ID '486685679562'. Notifications to this FCM Token will not be delivered over APNS.Be sure to re-retrieve the FCM token once the APNS device token is set.
8.9.0 - [Firebase/Messaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
8.9.0 - [Firebase/Messaging][I-FCM002022] APNS device token not set before retrieving FCM Token for Sender ID '486685679562'. Notifications to this FCM Token will not be delivered over APNS.Be sure to re-retrieve the FCM token once the APNS device token is set.
flutter: FCM Token: cIjfZ9AQC0zFhgiJYhryaK:APA91bFugDbFqUdS12fK-YBtteDe84NOWPsirlpeWp4RA4maJFE_KhgGdjLwNewgt-jkIcTLu-2nR3A4XCmBvRQPnxQlMZVFwD37OcAxlsQou8A0wIotbyPagyBl5T-SZHFCBh981vvM

```

</details>

```
[✓] Flutter (Channel stable, 2.10.2, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
[✓] Chrome - develop for the web
[✓] VS Code (version 1.62.0)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.
```

**Comment by cycloss** [comment]
Yes it compiles fine, and even functions fine wrt notifications sent from the fcm console; but the logs output a load of warnings which makes it seem like something has gone wrong.

**Comment by sstakoff** [comment]
I have same issue - with Cloud Messaging, I cannot run on real device unless I include a GoogleService-Info.plist file in addition to the dart only initialization

**Comment by combi-35** [comment]
Same for me, app gets stuck on launch with `Configuring the default Firebase app...`

**Comment by russellwheatley** [comment]
Hey @cycloss, not much we can do about those warnings. They're coming from the native SDK which is indicating (correctly if you're initializing from Dart) that the config file isn't there/app isn't initialized yet. The warnings all occur before the dart initialization has completed on the Dart side.

**Comment by cycloss** [comment]
@russellwheatley I see. It's kind of confusing for people seeing it though. It might be worth including something in the docs telling people to expect it. It looks like something has gone wrong when it hasn't, at least not for me. I think some people in this issue have different problems.

**Comment by russellwheatley** [comment]
@cycloss I see what you mean, I wonder if we should have a log in dart initialization saying something to the effect of: "Dart initialization has completed..". 

I guess this would give developers immediate feedback that it worked and negate previous warnings. I'll see what other contributors think.

**Comment by cycloss** [comment]
@russellwheatley sounds good. As long as there is something somewhere which draws developers attention to the fact that those warnings should be ignored if using `flutterfire configure`.

**Comment by sstakoff** [comment]
Hmm - for me it's more than just the messages - the app throws an exception and gets stuck - unusable. It only started happening after I added cloud messaging into the mix - was working fine with dart initialization with realtime database, authentication and cloud functions. No warnings printed with those. But once I added messaging, I was unable to proceed without actually installing a GoogleService-Info.plist file.
